nca/NetworkConfig/QueryOutputHandler.py

```diff
@@ -0,0 +1,383 @@
+    policies_pods: list[PoliciesWithCommonPods] = field(default_factory=list)  # used in DisjointnessQuery
+
+    def get_explanation_in_dict(self, explanation_description):
+        """
+        returns self type of OutputExplanation written in dict of description and policies_pods examples
+        :param explanation_description: the relevant description of this output explanation
+        :rtype: list[dict]
+        """
+        examples = []
+        for record in self.policies_pods:
+            examples.append({'policies': [record.first_policy_name, record.second_policy_name],
+                             'pods': record.common_pods})
+        return [{'description': explanation_description, 'examples': examples}]
+
+    def get_explanation_in_str(self, explanation_description):
+        """
+        returns self type of OutputExplanation written as str
+        :param explanation_description: the relevant description of this output explanation
+        :rtype: str
+        """
+        result = []
+        comma = ', '  # used for writing the list of pods joined by comma inline
+        for record in self.policies_pods:
+            result.append(f'{record.first_policy_type}_1: {record.first_policy_name}, '
+                          f'{record.second_policy_type}_2: {record.second_policy_name},'
+                          f' pods: {comma.join(record.common_pods)}')
+        return explanation_description + ':\n' + '\n'.join(result)
+
+
+@dataclass
+class PoliciesAndRulesExplanations:
+    # used in RedundancyQuery and EmptinessQuery: we may have lists of redundant/empty policies or
+    # maps of policies to redundant/empty ingress/egress rules indexes
+    policies_list: list[str] = field(default_factory=list)  # policy titles list, i.e. each element's form is:
+    # <policy_type> <policy_full_name>
+    policies_to_ingress_rules_dict: dict[str, list[int]] = field(default_factory=dict)
+    policies_to_egress_rules_dict: dict[str, list[int]] = field(default_factory=dict)
+
+    @staticmethod
+    def _get_policies_description(explanation_description):
+        return 'Policies' + explanation_description
+
+    @staticmethod
+    def _get_xgress_description(explanation_description, prefix='Ingress'):
+        return prefix + ' rules' + explanation_description
+
+    @staticmethod
+    def _add_rules_to_dict_explanation(xgress_description, xgress_rules_dict, prefix='ingress'):
+        """
+        returns dict object of policies to ingress/egress rules and its description
+        :param str xgress_description: the relevant description of this oof the given rules' dict
+        :param str prefix: indicates if rules types are ingress or egress
+        :return dict of description and pairs of policy and rules indexes
+        :rtype dict
+        """
+        rules = []
+        for key, value in xgress_rules_dict.items():
+            rules.append({'policy': key.split()[1], prefix + '_rules_indexes': [str(idx) for idx in value]})
```

Comment:
I would add a method to extract policy name from title instead of writing directly `key.split()[1]` , as it is less clear.